### PR TITLE
RavenDB-17066 - Fixed Assertion Failed at `string.IsNullOrEmpty(context.LastDatabaseChangeVector) == false`

### PR DESCRIPTION
**Pull request by lastav5**
https://issues.hibernatingrhinos.com/issue/RavenDB-17066

### CODE REVIEW

File: src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs [hunk 1]
```diff
@@ -1380,8 +1380,8 @@ protected override long ExecuteCmd(DocumentsOperationContext context)
                                  _replicationInfo.ReplicatedAttachmentStreams.Count == handledAttachmentStreams.Count,
                         "We should handle all attachment streams during WriteAttachment.");
 
-                    Debug.Assert(string.IsNullOrEmpty(context.LastDatabaseChangeVector) == false, $"database: {context.DocumentDatabase.Name}; context.LastDatabaseChangeVector = {context.LastDatabaseChangeVector ?? "null"};");
-
+                    Debug.Assert(context.LastDatabaseChangeVector != null);
```
Comment (ayende) [inline src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs]: How come that this happens?

Comment (lastav5) [reply src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs]: I put an explanation in YT: https://issues.hibernatingrhinos.com/issue/RavenDB-17066

Comment (ayende) [reply src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs]: Please add a comment here explaining why this is okay to be ""

Comment (lastav5) [reply src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs]: Done

**Review by ayende — APPROVED** [review]

**Review by karmeli87 — APPROVED** [review]

**Pull request merged**